vote/src/vote_account.rs

```diff
@@ -34,7 +34,7 @@ pub enum Error {
 #[derive(Debug)]
 struct VoteAccountInner {
     account: AccountSharedData,
-    vote_state: OnceLock<Result<VoteState, Error>>,
+    vote_state: VoteState,
```

Comment:
In contrast to the PR description, this change is not just removing the `OnceLock` but also dropping `Result` from the type.

These two changes should definitely have gone in separate PRs.
We really do not gain anything from mixing several independent changes into one PR, other than making it more likely that something subtle is missed out in the review or testing.

Dropping `Result` is indeed a functional change because now `VoteAccount::try_from(account_shared_data)` can fail in the ways that the previous code was not and this could have downstream consequences.
Is this functional change fine? I don't know, and having the change left out of the PR description and mixed in by other independent code changes does not help with scrutinizing that.

Reply:
> > These two changes should definitely have gone in separate PRs.

Once again - we're at 3 PRs out of 3 - a dismissive and condescending initial comment that makes me wonder if you've spent more than 10 seconds looking at the code. I invite you to think about what you're suggesting would have looked like:

- 1 PR with a ~5 line change removing OnceLock from the type declaration and the corresponding OnceLock::new() to initialize
- A second PR with all the changes in this PR minus the 5 lines above

So are you saying that you're not able to follow the bulk of the changes because of 5 lines?

Also it would be great if you could avoid saying _"definitely"_. To *me* it makes absolutely zero sense to split that way. The *whole reason* for having `Result` is that OnceLock *forced* us to have `Result`. If `OnceLock` goes, `Result` has absolutely no reason for being there anymore. Just removing `OnceLock` but *not* `Result` *makes no sense*. I'm not doing this change in a vacuum - I could care less about `OnceLock` - I need to put `VoteState` inside `VoteAccountInner` so I can deserializa in place. Maybe I should have added that code in this PR too to communicate the intent better and make reviewing easier! (In fact I did initially, but then I anticipated this conversation with you).

I understand that you seem to have a preference for micro PRs. I don't. In fact I _hate_ when PRs aren't atomic and contain all the logical changes that a changeset is trying to make, because then using the history to understand what some code does, which is something I all day every day, becomes harder.

> Dropping Result is indeed a functional change because now VoteAccount::try_from(account_shared_data) can fail in the ways that the previous code was not and this could have downstream consequences.
Is this functional change fine? I don't know

Did you read this? https://github.com/anza-xyz/agave/pull/2659#pullrequestreview-2251237535 and this https://github.com/anza-xyz/agave/pull/2271#discussion_r1692412042 ?

You don't know because you're not familiar with the code, or you're not convinced by the claim made above?

Reviews are hard work. I'd love for you to review this and the other PRs I tagged you in. If you don't have time or will to do it that's fine too, but then don't ask me to tag you for review.